### PR TITLE
add jasmine_node_test

### LINKED ISSUE #26: Provide jasmine_node_test rule
API should be like

```
jasmine_node_test(
    name = "test",
    srcs = ["foo.js", ":rule_that_produces_js"],
    data = [":some_nodejs_binary_rule"],
)
```

### PR DESCRIPTION
**Pull request by alexeagle**
fixes #26

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,18 +3,23 @@ jobs:
...
       - checkout
 
       - restore_cache:
           key: angular_bazel_example-{{ .Branch }}-{{ checksum "examples/rollup/yarn.lock" }}
       - run: bazel info release
+
       - run: bazel run @yarn//:yarn
-      - run: bazel build ...
-      - run: bazel test ...
+      - run: bazel build -- ... -examples/program/...
```
Comment (LucasSloan) [inline .circleci/config.yml]: Add a comment why the `-example/program/...`

File: examples/program/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        "node_modules/**/*.js",
+        "node_modules/**/*.d.ts",
+        "node_modules/**/*.json"
+    ]),
+    visibility = ["//visibility:public"],
+)
+
+jasmine_node_test(
+    name = "program",
+    srcs = glob(["*.js"]),
```
Comment (LucasSloan) [inline examples/program/BUILD.bazel]: As a developer looking at this example, I'd like an explanation of how jasmine_node_test handles sources vs. tests, which this glob doesn't really cover.

File: examples/program/WORKSPACE [hunk 3]
```diff
@@ -0,0 +1,21 @@
+workspace(name = "program_example")
```
Comment (LucasSloan) [inline examples/program/WORKSPACE]: Better name than program?  testable?

File: internal/devmode_js_sources.bzl [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  for d in ctx.rule.attr.deps:
+    if DevmodeJs in d:
+      sources += d[DevmodeJs].sources
+  sources += ctx.rule.attr.files
+  return [DevmodeJs(sources = sources)]
+
+def _devmode_js_sources_impl(ctx):
+  files = depset()
+  files += ctx.files.deps
+  # FIXME: needs TDD
```
Comment (LucasSloan) [inline internal/devmode_js_sources.bzl]: FIXME

File: examples/program/WORKSPACE (hunk 3 above)
Comment (alexeagle) [reply examples/program/WORKSPACE]: I'm planning to extend this example to cover a couple other missing things related to running code that you author under nodejs_binary (to reproduce problems currently seen only in repos that consume this) so I think this name will be okay.

File: .circleci/config.yml (hunk 1 above)
Comment (alexeagle) [reply .circleci/config.yml]: :+1:

File: examples/program/BUILD.bazel (hunk 2 above)
Comment (alexeagle) [reply examples/program/BUILD.bazel]: Good point, added a separate "library" rule as a filegroup, added that as a dep, so the srcs here are just *.spec.ts

**Comment by alexeagle** [comment]
PTAL

It occurs to me that we should never add another aspect for gathering up the manifest file, just re-use the same aspect that nodejs_binary has for locating sources. (In a subsequent PR)

**Pull request merged**